libraries/logging/iot_logging_task_dynamic_buffers.c

```diff
@@ -204,7 +226,14 @@ void vLoggingPrintf( const char * pcFormat,
         }
 
         xLength += ( size_t ) xLength2;
-        va_end( args );
+
+        /* Add newline characters if the message does not end with them.*/
+        ulFormatLen = strlen( pcFormat );
+
+        if( ( ulFormatLen >= 2 ) && ( strncmp( pcFormat + ulFormatLen, "\r\n", 2 ) != 0 ) )
```

Comment:
what if the buffer couldn't contain the complete formatted string?

Reply:
If the buffer couldn't complete the complete string, then this will overwrite the last 2 characters to still support a readable display of log messages on separate line 🙂